### DISCUSSION TITLE
How to change the "n" Parameter for one or more already existing databases

### DISCUSSION THREAD

**Discussion by Anubiso** [apache/couchdb#3226] [Q&A]
In the docs it is stated that you can enter the "n" and "q" values at the time of database creation.
https://docs.couchdb.org/en/latest/cluster/databases.html#creating-a-database

Is it possible to adapt these values once the database is already created? 
We switched from Single-Node Setup to 3 Node Setup and everything looks like it is working, besides the "n" parameter was not updated.

Current result of one /[db]/_db_info: 
  "cluster": {
        "q": 8,
        "n": 1,
        "w": 1,
        "r": 1
    },
Current result of one /[db]/_shards:
 "shards": {
        "00000000-1fffffff": [
            "couchdb@couchdb-0"
        ],
        "20000000-3fffffff": [
           "couchdb@couchdb-0"
        ],
        "40000000-5fffffff": [
            "couchdb@couchdb-0"
        ],
        "60000000-7fffffff": [
           "couchdb@couchdb-0"
        ],
        "80000000-9fffffff": [
            "couchdb@couchdb-0"
        ],
        "a0000000-bfffffff": [
            "couchdb@couchdb-0"
        ],
        "c0000000-dfffffff": [
            "couchdb@couchdb-0"
        ],
        "e0000000-ffffffff": [
            "couchdb@couchdb-0"
        ]
    }

Result of /_membership:
{
    "all_nodes": [
        "couchdb@couchdb-0",
        "couchdb@couchdb-1",
        "couchdb@couchdb-2",
    ],
    "cluster_nodes": [
        "couchdb@couchdb-0",
        "couchdb@couchdb-1",
        "couchdb@couchdb-2"
    ]
}

**Comment by wohali** [comment]
Most of this covered in https://docs.couchdb.org/en/stable/cluster/sharding.html , though it is not explicit that the same process is used for adding nodes to a database.

To increase `n` edit the `_dbs/<dbname>` document to add replicas on other nodes. You need to edit both the `by_node` and `by_range` sections, and it is recommended to modify the `changelog` as well. For your setup do not increase the number of replicas above 3 - only one replica can exist per node. It is then recommended to use the `/<db>/_sync_shards` endpoint.

If desired, you can increase `q` with the [Splitting Shards API](https://docs.couchdb.org/en/stable/cluster/sharding.html#splitting-shards).

Decreasing `q` requires replicating data into a database with a new `q` value, for that try a tool like https://github.com/neighbourhoodie/couch-continuum .
